### ISSUE TITLE
Sega Rally 2 has pixelated in-game graphics

### ISSUE THREAD

**Issue by antoniodesousa** [flyinghead/flycast#1006]
Platform / OS / Hardware: 
Windows 11 Pro 22H2
AMD Ryzen 9 6900HX with AMD Radeon 680M Graphics
32GB RAM + 2TB SSD

Github hash: [a21b79cfb](https://github.com/flyinghead/flycast/commit/a21b79cfbe9770d6718cb9ef6068de698e3d7f98)

**_Description of the Issue_**

Sega Rally 2 has pixelated in-game graphics. The track, the sky, the car... everything looks pixelated.

**_Screenshots_**

Sega Rally 2 running on Flycast:
![Sega Rally 2 - Flycast](https://user-images.githubusercontent.com/10920974/230452822-4e48134c-cdc8-41dc-b593-c48f4e23269b.png)

**Comment by Nejsplitter** [comment]
try using linear filtering in settings

**Comment by antoniodesousa** [comment]
Yes, forcing linear filtering does fix the issue.

**Comment by Alextended** [comment]
Forcing linear filtering does not completely fix the issue and/or introduces other issues as things still look very wrong in areas, for example in the "Muddy" tracks the rectangles which the tree leaves (not the palm trees, the other type) are drawn on and then their empty surface is made 100% transparent are visible. There are also text and other artifacts on some screens.

**Comment by Alextended** [comment]
I should maybe add the above happens with Vulkan in Windows 11 and updated RA//cores/Nvidia drivers with a GTX1080 and i7 3770J so I don't think it's my setup, just something very wrong with Sega Rally 2 rendering in general, which causes these issues when you try to fix the obvious glaring one, the unfiltered textures, with forcing them to be filtered, so you get the other issues then.

**Comment by Alextended** [comment]
Screenshots of the graphical issues encountered after forcing texture filtering to fix it not applying by default.
https://cdn.discordapp.com/attachments/252004626042191872/1196287943896207400/Sega_Rally_2-240115-045838.png
https://cdn.discordapp.com/attachments/252004626042191872/1196287941539016744/Sega_Rally_2-240115-045407.png
https://cdn.discordapp.com/attachments/252004626042191872/1196287940855332985/Sega_Rally_2-240115-045900.png
https://cdn.discordapp.com/attachments/252004626042191872/1196287942612746331/Sega_Rally_2-240115-045513.png
https://cdn.discordapp.com/attachments/252004626042191872/1196287943355150509/Sega_Rally_2-240115-045547.png

**Comment by Alextended** [comment]
This seems fixed now, I can force texture filtering with no issues (the screenshots above seem to have been deleted from discord server but they were showing glitching/artifacting in-race menus and pre-race car info screens and stuff, as well as the in-race trees in the Muddy environment having a transparent rectangle around their leaf textures). Hurray!

**Comment by flyinghead** [comment]
Glad to hear that forcing Linear filtering works.
However I'll keep this issue open since it shouldn't be necessary to change texture filtering for this game.

**Comment by mfigueirido** [comment]
This happens with Shenmue's skybox too:

![Shenmue (Europe) (En,Fr,De,Es) (Disc 1)-240929-131158](https://github.com/user-attachments/assets/f3849b3c-0382-4229-8e69-12e743295bd5)

Solved by forcing linear filtering:

![Shenmue (Europe) (En,Fr,De,Es) (Disc 1)-240929-131226](https://github.com/user-attachments/assets/a0a67ebf-96ce-4cc6-8be1-0f64631f20e9)

**Comment by mfigueirido** [comment]
Forcing linear filtering does create other issues though. All the user interface gets some line artifacts that look like gaps between tiles, very easy to spot on the inventory view:

![Shenmue (Europe) (En,Fr,De,Es) (Disc 1)-240929-171438](https://github.com/user-attachments/assets/aa3475f3-3bb4-4cd7-95b8-0f37b9393b0e)

This is how it looks without forcing linear:
![Shenmue (Europe) (En,Fr,De,Es) (Disc 1)-240929-171403](https://github.com/user-attachments/assets/2fa1b683-caf1-4ee9-aa20-e6f57e87f593)

**Comment by flyinghead** [comment]
@mfigueirido The pixelated skybox in Shenmue happens on real hardware too: https://youtu.be/ot_QlS1bjMw?si=EyFzkK2T6HvWdzg5&t=830

**Comment by mfigueirido** [comment]
You're right @flyinghead. The skybox looked so different to the other in-game graphics that I assumed it was a bug. My bad!

**Comment by antoniodesousa** [comment]
In the introduction cutscene, you can spot right away that there's no linear filtering used for the skybox. But yeah, I agree with you that it looks weird that everything else looks smooth except the sky. Probably it was left out to improve the performance in the game.

![Dreamcast Longplay - Shenmue (Part 1 of 8) - YouTube - Google Chrome 29_09_2024 19_31_39](https://github.com/user-attachments/assets/f6fbf3d9-06ef-4ca5-94fc-c0df85dcb296)

The image was taking of real hardware.
